filters/auth/oauth.go

```diff
@@ -173,50 +200,56 @@ func jsonGet(url *url.URL, auth string, doc interface{}) error {
 	return d.Decode(doc)
 }
 
-func newAuthClient(baseURL string) (*authClient, error) {
+func newAuthClient(baseURL string, timeout time.Duration) (*authClient, error) {
 	u, err := url.Parse(baseURL)
 	if err != nil {
 		return nil, err
 	}
-	return &authClient{url: u}, nil
+
+	quit := make(chan struct{})
```

Comment:
to call `close(quit)`, I would the following:

1. add a Close() method to the filterspec
2. call `close(spec.authClient.quit)` from this Close() method
3. for normal skipper operation do nothing, do not call close, because there's going to be only a single instance of the spec
4. for tests, always call Close() on the spec
5. document that in case someone uses filter spec as a library in a custom build, they need to call Close() if they may create an arbitrary number of spec instances

@addityasingh @szuecs what do you think?

Reply:
sounds good to me

Reply:
This approach looks clean to me, since we wouldn't need to pass down `quit` channel all the way down from `skipper` instance creation. And since the `Close()` is on `filterspec`, we wouldn't need to bother about the leaking goroutine as well. @szuecs Thoughts?

Reply:
Since the `FilterSpec` interface doesn't have the `Close` method, this approach wouldn't work. Check this https://play.golang.org/p/LkpQRoatvvn

Reply:
You might be able to type cast it. See: https://play.golang.org/p/BK0kM25jnjb

Reply:
It works in the playground, but not in the skipper test

Reply:
I see you found out. An interface can only be casted to, if you use the pointer IIRC.